osquery/remote/http/http_client.cpp

```diff
@@ -0,0 +1,317 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/http_client.h>
+#include <osquery/logger.h>
+
+namespace osquery {
+namespace http {
+
+/// This class is used to convert boost::system_exception
+/// to std::system_exception on freebsd since it is on boost-1.64.
+/// Idea taken from boost-1.65
+class adapted_category : public std::error_category {
+ public:
+  explicit adapted_category(boost::system::error_category const* pc)
+      : pc_(pc) {}
+
+  virtual const char* name() const noexcept {
+    return pc_->name();
+  }
+
+  virtual std::string message(int ev) const {
+    return pc_->message(ev);
+  }
+
+ private:
+  boost::system::error_category const* pc_;
```

Comment:
Can this be a unique pointer?

Reply:
No, since I am not handing over the ownership.